src/lib.rs

```diff
@@ -150,6 +151,32 @@ fn AggregateVerify(_py: Python, PKs: &PyList, messages: &PyList, signature: &PyB
     agg_sig.aggregate_verify(&msgs_refs, &pks_ref)
 }
 
+#[pyfunction]
+fn VerifyMultipleAggregateSignatures(_py: Python, SignatureSets: &PyList) -> bool {
+    let mut signature_sets: Vec<(AggregateSignature, AggregatePublicKey, Vec<u8>)> = Vec::new();
+    for set in SignatureSets {
+        let tuple: Vec<Vec<u8>> = match set.extract() {
+            Ok(_tuple) => _tuple,
+            Err(_) => return false,
+        };
+        let aggsig = match AggregateSignature::from_bytes(&tuple[0]) {
+            Ok(_aggsig) => _aggsig,
+            Err(_) => return false,
+        };
+        let aggkey = match PublicKey::from_bytes(&tuple[1]) {
+            Ok(_pubkey) => AggregatePublicKey::from_public_key(&_pubkey),
+            Err(_) => return false,
+        };
+        signature_sets.push((aggsig, aggkey, tuple[2].clone()));
+    }
+
+    let mut rng = &mut rand::thread_rng();
```

Comment:
@asanso instead of exposing a Python seed for the random number generator, I use the rust RNG here.

To faithfully wrap the verify_multiple_aggregate_signatures function, we suppose to export the rng argument. Without doing so, the internal working of this function is non-deterministic every time we run it.

But verify_multiple_aggregate_signatures is designed that the RNG argument should not determine the result with high probability. The function should accept correct SignatureSets and reject incorrect ones no matter how the internal RNG state is (with high probability).

Let me know your preference for which approach. I can fix it accordingly.